src/core/Akka.Tests/Actor/FSMTimingSpec.cs

```diff
@@ -8,223 +8,220 @@
-            ExpectMsg<Tick>(TimeSpan.FromMilliseconds(500));
-            Task.Delay(TimeSpan.FromMilliseconds(200));
+            fsm.Tell(State.TestCancelStateTimerInNamedTimerMessage);
+            fsm.Tell(Tick.Instance);
+            ExpectMsg(new Transition<State>(fsm, State.Initial, State.TestCancelStateTimerInNamedTimerMessage));
+            ExpectMsg<Tick>(500.Milliseconds());
+            Task.Delay(200.Milliseconds());
             Resume(fsm);
-            ExpectMsg(new FSMBase.Transition<State>(fsm, State.TestCancelStateTimerInNamedTimerMessage, State.TestCancelStateTimerInNamedTimerMessage2), FSMSpecHelpers.TransitionStateExpector<State>(), TimeSpan.FromMilliseconds(500));
-            fsm.Tell(new Cancel(), Self);
-            Within(TimeSpan.FromMilliseconds(500), () =>
+            ExpectMsg(new Transition<State>(fsm, State.TestCancelStateTimerInNamedTimerMessage, State.TestCancelStateTimerInNamedTimerMessage2), 500.Milliseconds());
+            fsm.Tell(Cancel.Instance);
+            Within(500.Milliseconds(), () =>
             {
                 ExpectMsg<Cancel>();
-                ExpectMsg(new FSMBase.Transition<State>(fsm, State.TestCancelStateTimerInNamedTimerMessage2, State.Initial), FSMSpecHelpers.TransitionStateExpector<State>());
-                return true;
+                ExpectMsg(new Transition<State>(fsm, State.TestCancelStateTimerInNamedTimerMessage2, State.Initial));
             });
         }
 
-        /// <summary>
-        /// receiveWhile is currently broken
-        /// </summary>
         [Fact]
         public void FSM_must_receive_and_cancel_a_repeated_timer()
         {
-            fsm.Tell(State.TestRepeatedTimer, Self);
-            ExpectMsg(new FSMBase.Transition<State>(fsm, State.Initial, State.TestRepeatedTimer), FSMSpecHelpers.TransitionStateExpector<State>());
-            var seq = ReceiveWhile(TimeSpan.FromSeconds(2), o =>
+            fsm.Tell(State.TestRepeatedTimer);
+            ExpectMsg(new Transition<State>(fsm, State.Initial, State.TestRepeatedTimer));
+            var seq = ReceiveWhile(2.Seconds(), o =>
             {
-                if(o is Tick) return o;
+                if (o is Tick)
+                    return o;
                 return null;
             });
-
-            Assert.Equal(5, seq.Count);
-            Within(TimeSpan.FromMilliseconds(500), () =>
+            seq.Should().HaveCount(5);
+            Within(500.Milliseconds(), () =>
             {
-                ExpectMsg(new FSMBase.Transition<State>(fsm, State.TestRepeatedTimer, State.Initial), FSMSpecHelpers.TransitionStateExpector<State>());
-                return true;
+                ExpectMsg(new Transition<State>(fsm, State.TestRepeatedTimer, State.Initial));
             });
         }
 
+        [Fact]
+        public void FSM_must_notify_unhandled_messages()
+        {
+            // EventFilter
```

Comment:
I'm assuming that's unrelated to this PR, and IIRC it's an area where we have some known bugs still anyway.